sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java

```diff
@@ -274,26 +278,35 @@ public void onNewUser(UserInformation userInfo) {
      */
     @Override
     public void onNetworkStateUpdated(final boolean connected) {
-        post(new Runnable() {
 
-            @Override
-            public void run() {
+        /* If device comes back online. */
+        if (connected) {
+            post(new Runnable() {
 
-                /* If device comes back online. */
-                if (connected) {
-                    for (PendingOperation po : mLocalDocumentStorage.getPendingOperations(Utils.getUserTableName())) {
-                        if (PENDING_OPERATION_CREATE_VALUE.equals(po.getOperation()) ||
-                                PENDING_OPERATION_REPLACE_VALUE.equals(po.getOperation())) {
-                            instanceCreateOrUpdate(po);
-                        } else if (PENDING_OPERATION_DELETE_VALUE.equals(po.getOperation())) {
-                            instanceDelete(po);
-                        } else {
-                            AppCenterLog.debug(LOG_TAG, String.format("Pending operation '%s' is not supported", po.getOperation()));
-                        }
-                    }
+                @Override
+                public void run() {
+                    processPendingOperations();
                 }
+            });
+        }
+    }
+
+    private void processPendingOperations() {
+        for (PendingOperation po : mLocalDocumentStorage.getPendingOperations(Utils.getUserTableName())) {
+            String outgoingId = Utils.getOutgoingId(po.getPartition(), po.getDocumentId(), po.getOperation());
+            if (mOutgoingCalls.contains(outgoingId)) {
```

Comment:
You can use `add` boolean result instead of `contains`. This makes 2 lookup otherwise.